### PR TITLE
#162725249 Integrate the application to use a postgresql database

### PR DESCRIPTION
**Pull request by mariamiah**
#### What does this PR do?
This PR is to integrate a PostgreSQL database into the application as its database engine

#### Description of Task to be completed?
Use PostgreSQL as it is open source, fully featured and optimized with production ready environments

#### How should this be manually tested?
- Fetch this branch `git fetch origin ch-integrate-a-database-162725249`
- Checkout to this branch `git checkout ch-integrate-a-database-162725249`
- Create a PostgreSQL database using `CREATE DATABASE preferred name` in the psql shell
- Enter the database credentials into the .env file following the convention in the sample_env file
- Start the virtual environment `source venv/bin/activate`
- Load the environment variables from the .env file `source .env`
- Run `python manage.py migrate`
- Start the development server using `python manage.py runserver`

#### What are the relevant pivotal tracker stories?
[#162725249](https://www.pivotaltracker.com/story/show/162725249)

### CODE REVIEW

File: authors/settings.py [hunk 1]
```diff
@@ -19,10 +8,10 @@
 # See https://docs.djangoproject.com/en/1.11/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = '7pgozr2jn7zs_o%i8id6=rddie!*0f0qy3$oy$(8231i^4*@u3'
+SECRET_KEY = config('SECRET_KEY')
 
 # SECURITY WARNING: don't run with debug turned on in production!
-DEBUG = True
+DEBUG = config('DEBUG', cast=bool)
```
Comment (MuhweziDeo) [inline authors/settings.py]: @mariamiah  I thought we were using os.getenv()  how does this work

**Review by armstrongsouljah — DISMISSED** [review]
LGTM

**Review by MuhweziDeo — DISMISSED** [review]
Other than the config method, we are using that I'm not well acquainted with, @mariamiah are good to go I see you set up to check state whether in production/staging. Good Job

File: authors/settings.py (hunk 1 above)
Comment (malfahad) [reply authors/settings.py]: @MuhweziDeo Have a look at this https://pypi.org/project/python-decouple/#why-not-just-use-environment-variables

**Review by malfahad — CHANGES_REQUESTED** [review]
Looks great. However, According to [python-decouple docs](https://pypi.org/project/python-decouple/#how-it-works)  

> Decouple always searches for Options in this order:
>  1. Environment variables;
>  2. Repository: ini or .env file;
>  3. default argument passed to config.

I suggest you rename `sample_env.txt` to have a `.env` extension.

**Comment by armstrongsouljah** [comment]
@malfahad The sample_env contains information that you need to set up your `.env` file locally. So basically, get the contents in the `sample_env.txt` and add them into a `.env` on your local machine, then replace the values with what's on your local machine. I hope that provides some clarity.

**Review by calvinpete — DISMISSED** [review]
@mariamiah Everything looks good to me

**Comment by malfahad** [comment]
@armstrongsouljah Thank you, it does. My case is solved

**Review by malfahad — DISMISSED** [review]

**Review by d-rita — COMMENTED** [review]
at what stage are we setting the definite names and values for the database setup? Just asking..

File: authors/settings.py (hunk 1 above)
Comment (MuhweziDeo) [reply authors/settings.py]: thanks problem solved

**Comment by armstrongsouljah** [comment]
@d-ritah database credentials for the local environment are to be added into a `.env` file thereby allowing the liberty of you connecting to the database you created on your local machine. Check the `sample_env.txt` for details to include in your `.env` file. I hope this all makes sense. Reach out incase it does not.

**Comment by ja-odur** [comment]
@mariamiah Please include the checkout step in the `how should this be manually tested` section of this PR.

**Review by armstrongsouljah — DISMISSED** [review]
LGTM

**Review by MuhweziDeo — DISMISSED** [review]

File: .travis.yml [hunk 2]
```diff
@@ -17,6 +17,8 @@ before-script:
 script:
     - python manage.py test
     - coverage run --source="authors" manage.py test&&coverage report -m
+env:
+    - SECRET_KEY = '7pgozr2jn7zs_o%i8id6=rddie!*0f0qy3$oy$ttt3'
```
Comment (ja-odur) [inline .travis.yml]: @mariamiah Please remove this `SECRET_KEY`. I have added it to the `Travis` environment.

Comment (mariamiah) [reply .travis.yml]: okay, was  testing if i would correct the travis error

**Review by MuhweziDeo — APPROVED** [review]
build passed on travis so your good to go

**Review by kdaglas — APPROVED** [review]
I have looked at it and its ok, secret-key is hidden, cool, good to go

**Review by calvinpete — APPROVED** [review]
@mariamiah Everything looks good to me

**Review by ja-odur — APPROVED** [review]
LGTM

**Pull request merged**